drivers/include/nrf24l01p.h

```diff
@@ -1,627 +1,1366 @@
+ * @brief   Possible values to configure the retransmission delay in ESB
  */
 typedef enum {
-    NRF24L01P_CRC_1BYTE = 0,    /**< encoding scheme generates 1 Byte redundancy */
-    NRF24L01P_CRC_2BYTE,        /**< encoding scheme generates 2 Bytes redundancy */
-} nrf24l01p_crc_t;
+    NRF24L01P_ARD_250us     = 0,    /**< 250 us */
+    NRF24L01P_ARD_500us     = 1,    /**< 500 us */
+    NRF24L01P_ARD_750us     = 2,    /**< 750 us */
+    NRF24L01P_ARD_1000us    = 3,    /**< 1000 us */
+    NRF24L01P_ARD_1250us    = 4,    /**< 1250 us */
+    NRF24L01P_ARD_1500us    = 5,    /**< 1500 us */
+    NRF24L01P_ARD_1750us    = 6,    /**< 1750 us */
+    NRF24L01P_ARD_2000us    = 7,    /**< 2000 us */
+    NRF24L01P_ARD_2250us    = 8,    /**< 2250 us */
+    NRF24L01P_ARD_2500us    = 9,    /**< 2500 us */
+    NRF24L01P_ARD_2750us    = 10,   /**< 2750 us */
+    NRF24L01P_ARD_3000us    = 11,   /**< 3000 us */
+    NRF24L01P_ARD_3250us    = 12,   /**< 3250 us */
+    NRF24L01P_ARD_3500us    = 13,   /**< 3500 us */
+    NRF24L01P_ARD_3750us    = 14,   /**< 3750 us */
+    NRF24L01P_ARD_4000us    = 15,   /**< 4000 us */
+    NRF24L01P_ARD_NUM_OF            /**< Number of possible values to configure the retransmission delay */
+} nrf24l01p_ard_t;
 
 /**
- * @brief   Defines the automatic retransmission delay defined from end of transmission
- * to start of next treansmission.
+ * @brief   Possible values to configure the CRC length
  */
 typedef enum {
-    NRF24L01P_RETR_250US = 0,   /**< retransmit delay is 250us */
-    NRF24L01P_RETR_500US,       /**< retransmit delay is 500us */
-    NRF24L01P_RETR_750US,       /**< retransmit delay is 750us */
-    NRF24L01P_RETR_1000US,      /**< retransmit delay is 1000us */
-    NRF24L01P_RETR_1250US,      /**< retransmit delay is 1250us */
-    NRF24L01P_RETR_1500US,      /**< retransmit delay is 1500us */
-    NRF24L01P_RETR_1750US,      /**< retransmit delay is 1750us */
-    NRF24L01P_RETR_2000US,      /**< retransmit delay is 2000us */
-    NRF24L01P_RETR_2250US,      /**< retransmit delay is 2250us */
-    NRF24L01P_RETR_2500US,      /**< retransmit delay is 2500us */
-    NRF24L01P_RETR_2750US,      /**< retransmit delay is 2750us */
-    NRF24L01P_RETR_3000US,      /**< retransmit delay is 3000us */
-    NRF24L01P_RETR_3250US,      /**< retransmit delay is 3250us */
-    NRF24L01P_RETR_3500US,      /**< retransmit delay is 3500us */
-    NRF24L01P_RETR_3750US,      /**< retransmit delay is 3750us */
-    NRF24L01P_RETR_4000US,      /**< retransmit delay is 4000us */
-} nrf24l01p_retransmit_delay_t;
-
-/**
- * @brief   Defines states for the nrf24l01+ transceiver
+    NRF24L01P_CRCO_0    = 1,    /**< 0 bytes CRC length */
+    NRF24L01P_CRCO_1    = 2,    /**< 1 byte CRC length */
+    NRF24L01P_CRCO_2    = 3,    /**< 2 bytes CRC length */
+    NRF24L01P_CRCO_NUM_OF       /**< Number of possible values to configure CRC length */
+} nrf24l01p_crco_t;
```

Comment:
Maybe `NRF24L01P_CRC_0` etc. and `nrf24l01p_crc_t` is easier to understand.

Reply:
Ok